libs/portals/core/src/components/PageLoader.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useRef } from 'react'
+import LoadingBar, { LoadingBarRef } from 'react-top-loading-bar'
+import { theme } from '@island.is/island-ui/theme'
+import { useNavigation } from 'react-router-dom'
+
+export const PageLoader = () => {
+  const { state } = useNavigation()
+  const ref = useRef<LoadingBarRef>(null)
+
+  useEffect(() => {
+    if (state === 'loading') {
+      ref.current?.continuousStart()
+    } else if (state === 'idle') {
+      ref.current?.complete()
+    }
+  }, [state])
+
+  const colorGradiant = `linear-gradient(90deg,
+    ${theme.color.blue400} 0%,
+    ${theme.color.blue600} 25%,
+    ${theme.color.purple400} 50%,
+    ${theme.color.roseTinted400} 75%,
+    ${theme.color.red400} 100%
+  )`
+
+  return <LoadingBar color={colorGradiant} ref={ref} />
```

Comment:
~Would be nice to move the visual part of this component to island-ui. Could even have a prop to start/stop to make it easier to integrate into our projects.

Reply:
Yeah I was thinking that as well but then I got lazy 😇 . I will do it now